src/main/java/com/joss/conductor/mobile/ConductorConfig.java

```diff
@@ -385,4 +372,60 @@ public boolean isAutoGrantPermissions() {
     public void setAutoGrantPermissions(boolean autoGrantPermissions) {
         this.autoGrantPermissions = autoGrantPermissions;
     }
+
+    public Platform getPlatformName() {
+        return platformName;
+    }
+
+    public String[] getCurrentSchemes() {
+        return currentSchemes;
+    }
+
+    public String getFullAppPath() {
+        if (appFile == null) {
+            return null;
+        }
+
+        Path path = Paths.get(appFile);
+        if (appFile.startsWith("sauce-storage:") || path.isAbsolute()) {
+            return appFile;
+        }
+        try {
+            URL url = new URL(appFile);
+            return appFile;
+        } catch(MalformedURLException e) {
+            // Ignore, this is expected to happen, parse as a regular path
+        }
+
+        return Paths.get(System.getProperty("user.dir"), path.normalize().toString()).normalize().toString();
+    }
+
+    public void setCustomCapabilities(Map<String, Object> customCapabilities) {
+        this.customCapabilities = customCapabilities;
+    }
+
+    public Map<String, Object> getCustomCapabilities() {
+        return customCapabilities;
+    }
+
+    private void putCustomCapabilities(Object keyCustomCapabilities) {
+        Map<String, Object> custom = new HashMap<>();
+        if (keyCustomCapabilities instanceof Map) {
+            for (String key : ((Map<String, Object>) keyCustomCapabilities).keySet()) {
+                Object value = ((Map) keyCustomCapabilities).get(key);
+                if (!(value instanceof Map)) {
+                    custom.put(key, value);
+                } else {
+                    throw new ClassCastException(String.format("%s cannot be of type Map", value));
```

Comment:
Why are explicitly forbidding `Map` type values?

Reply:
So what this prevents is the usage of nested maps in the `customCapabilities` field, like this:
```yaml
custom:
  customCapabilities:
    foo: foo
    fizz:
      bar: baz
      fazz: kazz
```
We might want to support this at some point, but currently I didn't see any need for supporting it, makes things a bit simpler.